src/dnx.common/include/tpa.h

```diff
@@ -3,5 +3,7 @@
 
 #pragma once
 
-BOOL CreateTpaBase(LPCTSTR** ppNames, size_t* pcNames, bool bNative);
-BOOL FreeTpaBase(const LPCTSTR* values);
+#include <vector>
+#include "xplat.h"
+
+std::vector<dnx::xstring_t> CreateTpaBase(bool bNative);
```

Comment:
Since TPA names are string literals this could just `std::vector<const dnx::xchar_t*>`